tensorflow/python/distribute/mirrored_strategy.py

```diff
@@ -502,9 +477,16 @@ def _experimental_make_numpy_dataset(self, numpy_input, session):
         numpy_input, self._host_input_device, session)
 
   def _experimental_distribute_datasets_from_function(self, dataset_fn,
+                                                      replication_mode,
```

Comment:
since this mode is only implemented for _experimental_distribute_datasets_from_function, if replica mode is passed to `_experimental_distribute_dataset`, that should throw an unsupported error, right?

Reply:
added unsupported check for calls to _experimental_distribute_dataset